crates/bevy_ecs/src/reflect.rs

```diff
@@ -114,12 +114,11 @@ impl ReflectComponent {
     }
 }
 
-impl<C: Component + Reflect + FromWorld> FromType<C> for ReflectComponent {
+impl<C: Component + Reflect + FromReflect> FromType<C> for ReflectComponent {
     fn from_type() -> Self {
         ReflectComponent {
             insert: |world, entity, reflected_component| {
-                let mut component = C::from_world(world);
-                component.apply(reflected_component);
+                let component = C::from_reflect(reflected_component).unwrap();
```

Comment:
Since `FromReflect::from_reflect` doesn't currently return a `Result`, could we change this `unwrap` to an `expect` (or `unwrap_or_else(|| panic!(...))`)?

Same for similar calls below.